packages/python-sdk/e2b/sandbox_sync/main.py

```diff
@@ -149,14 +152,29 @@ def __init__(
                 domain=domain,
                 debug=debug,
                 request_timeout=request_timeout,
+                secure=secure or False,
                 proxy=proxy,
             )
             self._sandbox_id = response.sandbox_id
             self._envd_version = response.envd_version
 
-        self._envd_api_url = f"{'http' if self.connection_config.debug else 'https'}://{self.get_host(self.envd_port)}"
+            if response.envd_access_token is not None and not isinstance(response.envd_access_token, Unset):
+                self._envd_access_token = response.envd_access_token
+                connection_headers["X-Access-Token"] = response.envd_access_token
+            else:
+                self._envd_access_token = None
 
         self._transport = TransportWithLogger(limits=self._limits, proxy=proxy)
+        self._connection_config = ConnectionConfig(
+            api_key=api_key,
+            domain=domain,
+            debug=debug,
+            request_timeout=request_timeout,
+            headers=connection_headers,
+            proxy=proxy,
+        )
+
+        self._envd_api_url = f"{'http' if self.connection_config.debug else 'https'}://{self.get_host(self.envd_port)}"
         self._envd_api = httpx.Client(
```

Comment:
can't you set the headers here already? you wouldn't need to pass it around Commands, Filesystem, Pty methods

Reply:
```python
            if response.envd_access_token is not None and not isinstance(
                response.envd_access_token, Unset
            ):
                self._envd_access_token = response.envd_access_token
                connection_headers["X-Access-Token"] = response.envd_access_token
            else:
                self._envd_access_token = None
```

setting it on line 163 in same method call.

Reply:
i meant 
```py
        self._envd_api = httpx.AsyncClient(
            base_url=self.envd_api_url,
            transport=self._transport,
            headers=self._connection_config.headers,
        )
```